conda_smithy/update_cb3.py

```diff
@@ -215,19 +221,22 @@ def update_cb3(recipe_path, conda_build_config_path):
                     need_mingw_c = True
                 continue
             if req_rendered == 'cython' and not (need_c or need_cxx or need_f):
-                messages['Found cython requirement. Adding compiler'] = True
-                need_c = True
-            if req in ['ninja', 'jom', 'cmake', 'automake', 'autoconf', 'libtool',
+                if 'c' not in compilers:
+                    messages['Found cython requirement. Adding compiler'] = True
+                    need_c = True
+            if req in (['ninja', 'jom', 'cmake', 'automake', 'autoconf', 'libtool',
                        'make', 'pkg-config', 'automake-wrapper', 'posix', 'm4'] \
                     or req.startswith("{{p") or req.startswith("m2-") \
                     or (req_rendered in ['perl', 'texlive-core', 'curl', 'openssl', 'tar', 'gzip', 'patch']
-                        and section == 'build' and req_rendered not in reqs['run']):
+                        and req_rendered not in reqs['run']))
+                    and section == 'host':
```

Comment:
cb2 recipes don't have a host section

Reply:
Right but update-cb3 changes build to host at the beginning. I guess we could just say if there is a host section do nothing since it's already cb3, but I was trying to allow for the possibility that something went wrong along the way and there is a host section but missing needed compilers or other dependencies that should be in the build section. Though maybe that would never happen since CI would fail?

Reply:
> Right but update-cb3 changes build to host at the beginning.

No, it doesn't. It's only changed in `change_lines`